HyperVExtension/src/HyperVExtension/Models/VirtualMachineCreation/VMGalleryVMCreationOperation.cs

```diff
@@ -0,0 +1,247 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using HyperVExtension.Common;
+using HyperVExtension.Exceptions;
+using HyperVExtension.Helpers;
+using HyperVExtension.Models.VMGalleryJsonToClasses;
+using HyperVExtension.Providers;
+using HyperVExtension.Services;
+using Microsoft.Windows.DevHome.SDK;
+using Windows.Foundation;
+using Windows.Storage;
+
+namespace HyperVExtension.Models.VirtualMachineCreation;
+
+public delegate VMGalleryVMCreationOperation VmGalleryCreationOperationFactory(VMGalleryCreationUserInput parameters);
+
+/// <summary>
+/// Class that represents the VM gallery VM creation operation.
+/// </summary>
+public sealed class VMGalleryVMCreationOperation : IVMGalleryVMCreationOperation
+{
+    private const string ComponentName = nameof(VMGalleryVMCreationOperation);
+
+    private readonly IArchiveProviderFactory _archiveProviderFactory;
+
+    private readonly IHyperVManager _hyperVManager;
+
+    private readonly IDownloaderService _downloaderService;
+
+    private readonly string _tempFolderSaveLocation = Path.GetTempPath();
+
+    private readonly IStringResource _stringResource;
+
+    private readonly IVMGalleryService _vmGalleryService;
+
+    private readonly VMGalleryCreationUserInput _userInputParameters;
+
+    public CancellationTokenSource CancellationTokenSource { get; private set; } = new();
+
+    private readonly object _lock = new();
+
+    public bool IsOperationInProgress { get; private set; }
+
+    public bool IsOperationCompleted { get; private set; }
+
+    public CreateComputeSystemResult? ComputeSystemResult { get; private set; }
+
+    public StorageFile? ArchivedFile { get; private set; }
+
+    public VMGalleryImage Image { get; private set; } = new();
+
+    public event TypedEventHandler<ICreateComputeSystemOperation, CreateComputeSystemActionRequiredEventArgs>? ActionRequired = (s, e) => { };
```

Comment:
This is not used yet, but when we'll use it, we'll likely want to know if there is a subscriber to this event and having it initialized to null will show that there are no subscribers.

Reply:
Interestingly got a build error that it wasn't used. We can probably suppress it but figured for now I'll just leave it as is thanks